docs/proposals/20210325-imagepredownload-during-inplace-update.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Proposal Template
+authors:
+- "@FillZpp"
+reviewers:
+- "@Fei-Guo"
+- "@furykerry"
+creation-date: 2021-03-26
+last-updated: 2021-03-26
+status: implementable
+---
+
+# Pre-download image during in-place update
+
+If CloneSet is going to update Pods in-place,
+the controller could pre-download the new image on the Nodes of all Pods in the CloneSet.
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a proposal and for highlighting
+any additional information provided beyond the standard proposal template.
+[Tools for generating](https://github.com/ekalinin/github-markdown-toc) a table of contents from markdown are available.
+
+- [Pre-download image during in-place update](#pre-download-image-during-in-place-update)
+  - [Table of Contents](#table-of-contents)
+  - [Motivation](#motivation)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+    - [Requirements](#requirements)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Implementation History](#implementation-history)
+
+## Motivation
+
+If the image has been pre-downloaded on Nodes, we can significantly accelerate the speed of in-place update,
+for Kubelet has not to pull the image again.
+
+## Proposal
+
+Add a new feature-gate: `PreDownloadImageForInPlaceUpdate`.
+
+If it is opened, CloneSet will create a `ImagePullJob` when it starts a new in-place update:
```

Comment:
multiple ImagePullJob maybe needed, if there are multiple containers in the pod.